lib/errors.js

```diff
@@ -198,15 +198,20 @@ exports.process = function (errors, object) {
         return override;
     }
 
-    const error = new Error(message);
-    error.isJoi = true;
-    error.name = 'ValidationError';
-    error.details = details;
-    error._object = object;
-    error.annotate = internals.annotate;
-    return error;
+    return new exports.ValidationError(message, details, object);
 };
 
+exports.ValidationError = class ValidationError extends Error {
+    constructor(message, details, object) {
+
+        super(message);
+        this._object = object;
+        this.annotate = internals.annotate;
```

Comment:
I think `annotate`, `isJoi` and `name` can be part of the prototype as they don't change between errors. As node still doesn't support class properties I'd suggest attaching it to the prototype like `exports.ValidationError.prototype.isJoi = true` right after.

I have a doubt about `name` since it's part of `Error` so if you can't for this one, leave it be.